static/app/components/replays/replayContext.tsx

```diff
@@ -381,25 +307,27 @@ export function Provider({
         setIsPlaying(false);
       }
     },
-    [startTimeOffsetMs, durationMs, getCurrentTime, isPlaying]
+    [startTimeOffsetMs, durationMs, getCurrentPlayerTime, isPlaying]
   );
 
   const setCurrentTime = useCallback(
     (requestedTimeMs: number) => {
-      privateSetCurrentTime(requestedTimeMs);
+      privateSetCurrentTime(requestedTimeMs + startTimeOffsetMs);
       clearAllHighlights();
     },
-    [privateSetCurrentTime, clearAllHighlights]
+    [privateSetCurrentTime, startTimeOffsetMs, clearAllHighlights]
   );
 
   const applyInitialOffset = useCallback(() => {
+    const offsetMs = (initialTimeOffsetMs?.offsetMs ?? 0) + startTimeOffsetMs;
+
     if (
       !didApplyInitialOffset.current &&
-      initialTimeOffsetMs &&
+      (initialTimeOffsetMs || offsetMs) &&
```

Comment:
is it a problem that `offsetMs` will be a truthy value whenever `startTimeOffsetMs` is set, ie whenever it's a clipped replay

Reply:
This is intended since clips will always need to start playback at a point in time other than 0